dev-packages/application-package/src/application-package.ts

```diff
@@ -120,6 +119,27 @@ export class ApplicationPackage {
         return this._extensionPackages;
     }
 
+    /**
+     * All entry points defined by the installed Theia extensions.
+     *
+     * Map of entry point name to script import path.
+     */
+    get extensionEntryPoints(): ReadonlyMap<string, string> {
+        if (!this._extensionEntryPoints) {
```

Comment:
I typically frown upon lazy initialization for two reasons:

1. If the init is fast, we should just do it at object construction
2. If it is slow, we should make that explicit by returning a promise. A method that is fast or slow arbitrarily is usually not what we want.
3. If you don't control when the init is done, the init might be invoked before it's prerequisites are fulfilled.

Reply:
Ok, I just aligned with what was already there. VS Code have a `@memoize` decorator for that kind of stuff, so idk really. I'd keep the code that way for now, until we decide we should change the rest too?